ext/dcos-installer/dcos_installer/installer_analytics.py

```diff
@@ -2,33 +2,36 @@
 import os
 import uuid
 
+from dcos_installer import backend
+
 
 class InstallerAnalytics():
-    def __init__(self, config):
+    def __init__(self):
         self.uuid = self.new_uuid()
-        self.bootstrap_version = os.getenv("BOOTSTRAP_ID")
-        self.customer_key = config["customer_key"] if 'customer_key' in config else ""
+        self.bootstrap_version = os.getenv("BOOTSTRAP_ID", "")
+        self.customer_key = ""
         self.source = "onprem"  # If you're using the installer you're not in AWS or Azure
 
-    def send(self, action, install_method, num_errors):
+    def send(self, action="", install_method="", num_errors=0):
         """Sends analytics track data to segmentIO.
         variant: string | open or enterprise
         action: string | preflight, deploy, or postflight
         install_method: string | gui, cli or advanced
         """
         analytics.write_key = "39uhSEOoRHMw6cMR6st9tYXDbAL3JSaP"
 
+        # We set customer key from config to avoid loading the config during class init
+        self.customer_key = backend.get_config()["customer_key"] if 'customer_key' in backend.get_config() else ""
+
         analytics.track(user_id=self.customer_key, anonymous_id=self.uuid, event=action, properties={
             "install_id": self.uuid,
             "bootstrap_id": self.bootstrap_version,
-            "provider": self.source,
-            "source": "installer",
+            "source": self.source,
             "install_method": install_method,
             "stage": action,
             "errors": num_errors,
-            "customer_key": self.customer_key,
         })
         analytics.flush()
 
     def new_uuid(self):
-        return str(uuid.uuid4())
```

Comment:
str definitely seems much more likely to be the format we want it in. (then things show up as a uuid, instead of the bits that make up the uuid)